sdk/Sdk.Generators/ExtensionStartupRunnerGenerator.cs

```diff
@@ -0,0 +1,212 @@
+
+        /// <summary>
+        /// Fully qualified name of the above "WorkerExtensionStartupAttribute" attribute.
+        /// </summary>
+        private string attributeTypeFullName = "Microsoft.Azure.Functions.Worker.Core.WorkerExtensionStartupAttribute";
+
+        /// <summary>
+        /// Fully qualified name of the base type which extension startup classes should implement.
+        /// </summary>
+        private string startupBaseClassName = "Microsoft.Azure.Functions.Worker.Core.WorkerExtensionStartup";
+
+        public void Execute(GeneratorExecutionContext context)
+        {
+            var startupTypes = GetExtensionStartupTypes(context);
+
+            if (!startupTypes.Any())
+            {
+                return;
+            }
+
+            SourceText sourceText;
+            using (var stringWriter = new StringWriter())
+            using (var indentedTextWriter = new IndentedTextWriter(stringWriter))
+            {
+                indentedTextWriter.WriteLine("// <auto-generated/>");
+                indentedTextWriter.WriteLine("using System;");
+                indentedTextWriter.WriteLine("using Microsoft.Azure.Functions.Worker.Core;");
+                WriteAssemblyAttribute(indentedTextWriter);
+                indentedTextWriter.WriteLine("namespace Microsoft.Azure.Functions.Worker");
+                indentedTextWriter.WriteLine("{");
+                indentedTextWriter.Indent++;
+                WriteStartupCodeExecutorClass(indentedTextWriter, startupTypes);
+                indentedTextWriter.Indent--;
+                indentedTextWriter.WriteLine("}");
+
+                indentedTextWriter.Flush();
+                sourceText = SourceText.From(stringWriter.ToString(), encoding: Encoding.UTF8);
+            }
+
+            // Add the source code to the compilation
+            context.AddSource($"WorkerExtensionStartupCodeExecutor.g.cs", sourceText);
+        }
+
+        /// <summary>
+        /// Gets the extension startup implementation type info from each of the participating extensions.
+        /// Each entry in the return type collection includes full type name 
+        /// & a potential error message if the startup type is not valid.
+        /// </summary>
+        private IEnumerable<(string TypeName, string? Error)> GetExtensionStartupTypes(GeneratorExecutionContext context)
+        {
+            List<(string TypeName, string? Error)>? typeInfoList = null;
+
+            // Extension authors should decorate their assembly with "WorkerExtensionStartup" attribute
+            // if they want to participate in startup.
+            foreach (var assembly in context.Compilation.SourceModule.ReferencedAssemblySymbols)
+            {
+                var extensionStartupAttribute = assembly.GetAttributes()
+                                                        .FirstOrDefault(a => a.AttributeClass?.Name == attributeTypeName &&
```

Comment:
nit: use string.Equals with the appropriate comparison in both string checks here

Reply:
Switched to the overload which takes StringComparison and passing `StringComparison.Ordinal` . (No need to check case insensitive check here).

Interestingly, [in my microbenchmarks](https://gist.github.com/kshyju/dc78a1f5ef9e8da40c1b4985463f79e6), the `Equals` overload without comparison type parameter does better by a tiny fraction.